### ISSUE TITLE
Make other optimizers available e.g. prodigy

### ISSUE THREAD

**Issue by jan-forest** [jan-forest/autoencodix_package#191]
Currently, we use AdamW as PyTorch optimizer. Users may benefit from usage of other optimizers. In particular, Prodigy https://github.com/konstmish/prodigy might be a good addition since it does not require learning rate optimization.
